File: Programmers/Level2/수식최대화/수식최대화.py
import re
from itertools import permutations

def solution(expression):
    answer = []
    operators = list(permutations(['*', '+', '-'], 3))
    expression = re.split('([*+-])', expression)
    
    for operator in operators:
        exp = expression[:]
    
        for op in operator:
            while op in exp:
                idx = exp.index(op)
                exp[idx - 1] = str(eval(exp[idx - 1] + op + exp[idx + 1]))
                del exp[idx:idx + 2]
        answer.append(abs(int(exp[0])))
        
    return max(answer)

File: Programmers/Level2/수식최대화/수식최대화.js
function solution(expression) {
    let answer = [];
    const operators = [
        ["+", "-", "*"],
        ["+", "*", "-"],
        ["-", "+", "*"],
        ["-", "*", "+"],
        ["*", "+", "-"],
        ["*", "-", "+"],
    ];
    expression = expression.split(/([\*\+\-])/g);

    for (let operator of operators) {
        let exp = [...expression];

        for (let op of operator) {
            while (exp.includes(op)) {
                let idx = exp.indexOf(op);
                exp[idx - 1] = String(eval(exp[idx - 1] + op + exp[idx + 1]));
                exp.splice(idx, 2);
            }
        }
        answer.push(Math.abs(parseInt(exp[0])));
    }

    return Math.max(...answer);
}
